gamestonk_terminal/cryptocurrency/discovery/coinpaprika_model.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+from gamestonk_terminal.cryptocurrency.coinpaprika_helpers import (
+    ENDPOINTS,
+    PaprikaSession,
+)
+
+
+session = PaprikaSession()
```

Comment:
Probably we could have this inside the commands?

Reply:
@DidierRLopes 

I agree to not keep it initialized in model. I tried here few things, but without good results. 

One thing I triend was injection of PaprikaSession from Controller to `coinpaprika_model`:

```
class CryptoController:
    def __init__(self, coin=None, source=None):
        """CONSTRUCTOR"""

        self._dd_parser = argparse.ArgumentParser(add_help=False, prog="dd")

        self.current_coin = coin
        self.current_currency = None
        self.current_df = pd.DataFrame()
        self.source = source

        self.CHOICES.extend(self.SPECIFIC_CHOICES[self.source])

        self._dd_parser.add_argument("cmd", choices=self.CHOICES)

        if not coinpaprika_model.session:
            coinpaprika_model.session = PaprikaSession()
```

And in the same time looked  `coinpaprika_model` looked like here:
```
"""CoinPaprika model"""
__docformat__ = "numpy"

from typing import Optional, Any
import pandas as pd

session = None

def search(
    query: str, category: Optional[Any] = None, modifier: Optional[Any] = None
) -> pd.DataFrame:
    ...
    return pd.DataFrame(results)
```

So it was an idea to just inject session to given _model.py at the moment of initialization CryptoController, but this is not working. I think it's because some order of imports in other .py files. When I use any command related to CoinPaprika I get that this `session` variable is still `None`
But I am not sure. Maybe there is a way to set this variable as a `global` , but I am not familiar with that, so hard to say. Maybe you have more knowledge with that. 

Another Idea that I have, to just wrap all methods in coinpaprika_model.py into a class which will have in `__init__` parameter session which I will provide in Controller classes. But it's just an idea, don't know if it's right approach. 

Maybe you have any other idea for that ? Would be great, cause I feel a little stuck with it :D

Reply:
Thanks for this explanation, I'm ok to leave this as is for now. And once we go into refactoring we can look at ways to tackle this. We can have a joined session with @jmaslek to brainstorm ideas to achieve the best architecture possible.

This will lead to have a strong foundation, and a more robust code going forward!